installer/installer_centos.sh

```diff
@@ -1,3 +1,141 @@
+if grep 'VERSION="7' /etc/os-release; then
+    ${sudo_cmd} cp /etc/yum/CentOS-Base.repo /etc/yum/CentOS-Base.repo.bak
+    ${sudo_cmd} sed -e 's|^mirrorlist=|#mirrorlist=|g' \
+    -e 's|^#baseurl=http://mirror.centos.org/centos|baseurl=https://mirrors.tuna.tsinghua.edu.cn/centos|g' \
+    -i.bak \
+    /etc/yum.repos.d/CentOS-*.repo | ${sudo_cmd} tee /etc/yum/CentOS-Base.repo || exit 1
+    ${sudo_cmd} yum update || exit 1
+elif grep 'VERSION="8' /etc/os-release;then
+    ${sudo_cmd} cp /etc/yum.repos.d/CentOS-Linux-BaseOS.repo /etc/yum.repos.d/CentOS-Linux-BaseOS.repo.bak
+    ${sudo_cmd} sed -e 's|^mirrorlist=|#mirrorlist=|g' \
+    -e 's|^#baseurl=http://mirror.centos.org/\$releasever|baseurl=https://mirrors.tuna.tsinghua.edu.cn/centos|g' \
+    -i.bak \
+    /etc/yum.repos.d/CentOS-*.repo | ${sudo_cmd} tee /etc/yum/CentOS-Base.repo || exit 1
+    ${sudo_cmd} yum update || exit 1
+else
+    log "only support centos7 and centos8,change the source by yourself"
+fi
+
+# install sudo
+if !  ${sudo_cmd} --version;then
+    ${sudo_cmd} yum install -y sudo
+fi
+
+# basic tools
+${sudo_cmd} yum install -y curl wget tar ripgrep || exit 1
+
+
+# this is for add-yum-repository
+sudo yum install -y yum-utils || exit 1
+if grep 'VERSION="7 (Core)' /etc/os-release; then
+    sudo yum install -y epel-release || exit 1
+fi
+# install fish 3
+if ! fish --version;then
+
+    url="https://download.opensuse.org/repositories/shells:fish:release:3/CentOS_"
+    if grep 'VERSION="8"' /etc/os-release; then
+        url="${url}8/shells:fish:release:3.repo"
+     # centos6
+    elif grep 'VERSION="7 (Core)' /etc/os-release; then
+        url="${url}7/shells:fish:release:3.repo"
+    else 
+        url=""
+    fi
+    if [ -z "$url" ]; then
+        echo "only support CentOS 8 and 7"
+        exit 1
+    fi
+
+    sudo wget "$url" -O /etc/yum.repos.d/fish.repo || exit 1
+    sudo yum install -y fish || exit 1
+    
+    # get path of fish command
+    fish_cmd=$(command -v fish)
+
+    # check fish is include in /etc/shells?
+    if ! grep "^$fish_cmd\$" /etc/shells; then
+        echo "$fish_cmd" | sudo tee -a /etc/shells || exit 1
+    fi
+    # we only set fish as the login shell for non-root users
+    if [ $UID != 0 ]; then
+        chsh -s "$fish_cmd" || exit 1
+    fi
+fi
+
+# update vim to vim-9
+if ! vim --version | grep '9\.0'; then
+    sudo yum install -y vim-enhanced || exit 1
+fi 
+# ctags are used for the outlooks of markdown files
+sudo yum install -y ctags || exit 1
+
+# install vim plugins
+# after installation you may need quit vim manually by using :qa
+# this will fail to install LeaaderF and markdown-preview,
+# but don't worry, the script will solve this at last
+vim +PlugInstall 
+
+
+# some tool for development 
+sudo yum groupinstall -y "Development Tools" || exit 1
+sudo yum install -y cmake gdb python3-devel python3-pip pandoc || exit 1
+
+# installation for ShellCheck
+sudo yum install -y ShellCheck || exit 1
+
+# clang
```

Comment:
this will not install `clang-format`. Install `clang-tidy` and `clang-format` seperatedly rather than installing `clang-tools-extra` and You should install `clangd` rather than `clang`.